torchtitan/train.py

```diff
@@ -387,13 +387,10 @@ def forward_backward_step(
                     (labels, []) if self.pp_has_last_stage else (None, None)
                 )
                 if self.pp_has_first_stage:
-                    self.pp_schedule.step(
-                        inputs, target=targets, losses=losses, input_batch=inputs
-                    )
+                    # TODO: need to update kwargs for different models?
+                    self.pp_schedule.step(inputs, target=targets, losses=losses)
```

Comment:
Llama3/4 [forward](https://github.com/pytorch/torchtitan/blob/01f4e50228fb4a304d17b5b5c5edfe39cc960000/torchtitan/models/llama3/model/model.py#L401) is slightly different from DeepSeekv3 [forward](https://github.com/pytorch/torchtitan/blob/0a6ab71b76d9831b81b27296522b230bed44ec3c/torchtitan/models/deepseek_v3/model/model.py#L361). I think the `input_batch` argument is needed for `flex_attn`.

What is the best way to go about this, should we create a new Trainer class for DeepSeekV3 that overrides `forward_backward_step`?